### PR TITLE
magento/devdocs#8873: Add clarification about required parameter for date validation

### PR DESCRIPTION
**Pull request by EvheniiaKolkhiieva**
## Purpose of this pull request

Fix pull request fixes https://github.com/magento/devdocs/issues/8873

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.4/javascript-dev-guide/widgets/widget_calendar.html

## Links to Magento source code

-  https://github.com/magento/magento2/blob/2.4/lib/web/mage/validation.js#L1046

### CODE REVIEW

File: src/guides/v2.3/javascript-dev-guide/widgets/widget_calendar.md [hunk 1]
```diff
@@ -41,6 +41,28 @@ require([
 });
 ```
 
+## Validate the calendar widget
+
+To validate the calendar widget, required **dateFormat** parameter should be specified, as can be seen from the validate-date rule initialisation:
```
Comment (atwixfirster) [inline src/guides/v2.3/javascript-dev-guide/widgets/widget_calendar.md:46]: @EvheniiaKolkhiieva  could you please highlight a validate-date - `validate-date`.

Thank you!

**Comment by atwixfirster** [comment]
@EvheniiaKolkhiieva , thank you for your contribution. 

A small notice here: your PR totally makes sense for Magento 2.4.0 and newest versions. 

Please add your notices to `2.4` version of this document.

Thank you!

File: src/guides/v2.3/javascript-dev-guide/widgets/widget_calendar.md [hunk 2]
```diff
@@ -41,6 +41,28 @@ require([
 });
 ```
 
+## Validate the calendar widget
+
+To validate the calendar widget, required **dateFormat** parameter should be specified, as can be seen from the **validate-date** rule initialisation:
+
+```javascript
+'validate-date': [
+            function (value, params, additionalParams) {
```
Comment (rogyar) [inline src/guides/v2.3/javascript-dev-guide/widgets/widget_calendar.md:50]: Please, make sure we use the correct indentation (number of spaces before each line of code) here. 
Let me provide a correct version: 

```javascript
'validate-date': [
    function (value, params, additionalParams) {
        var test = moment(value, utils.convertToMomentFormat(additionalParams.dateFormat));

        return $.mage.isEmptyNoTrim(value) || test.isValid();
    },
    $.mage.__('Please enter a valid date.')
]
```

File: src/guides/v2.3/javascript-dev-guide/widgets/widget_calendar.md [hunk 3]
```diff
@@ -41,6 +41,28 @@ require([
...
+                var test = moment(value, utils.convertToMomentFormat(additionalParams.dateFormat));
+
+                return $.mage.isEmptyNoTrim(value) || test.isValid();
+            },
+            $.mage.__('Please enter a valid date.')
+
```
Comment (rogyar) [inline src/guides/v2.3/javascript-dev-guide/widgets/widget_calendar.md:56]: Also, please, remove this extra empty line here.

**Review by rogyar — CHANGES_REQUESTED** [review]
Hi @EvheniiaKolkhiieva. Great job! Please, check my suggestions below

**Comment by EvheniiaKolkhiieva** [comment]
Thank you @atwixfirster and @rogyar for your attention!

I have implemented the fixes you suggested, please check!

Thank you!

**Review by atwixfirster — APPROVED** [review]
Approve from my side!

Thank you, @EvheniiaKolkhiieva

**Review by rogyar — APPROVED** [review]

**Comment by dobooth** [comment]
@magento import code to magento-devdocs/devdocs

**Comment by magento-engcom-team** [comment]
@dobooth the branch with code successfully imported into`magento-devdocs/devdocs` repository. Branch name: `imported-magento-devdocs-8874`.

**Pull request merged**